KD_Lib/KD/vision/CSKD/cskd.py

```diff
@@ -0,0 +1,172 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+
+import matplotlib.pyplot as plt
+from copy import deepcopy
+
+from KD_Lib.KD.common import BaseClass
+
+
+class KDLoss(nn.Module):
+    def __init__(self, temp_factor):
+        super(KDLoss, self).__init__()
+        self.temp_factor = temp_factor
+        self.kl_div = nn.KLDivLoss(reduction="sum")
+
+    def forward(self, input, target):
+        log_p = torch.log_softmax(input / self.temp_factor, dim=1)
+        q = torch.softmax(target / self.temp_factor, dim=1)
+        loss = self.kl_div(log_p, q) * (self.temp_factor ** 2) / input.size(0)
+        return loss
```

Comment:
you can prolly shift this to CSKD class itself, we currently use `calculate_kd_loss` for those who have this loss function.

Reply:
Okay